src/lib/dom-api-mutation.html

```diff
@@ -0,0 +1,280 @@
+          var c$ = this.domApi.childNodes;
+          if (c$.length) {
+            for (var i=0, c; (i < c$.length) && (c=c$[i]); i++) {
+              this._addedNodes.push(c);
+            }
+            this._scheduleNotify();
+          }
+        }
+      },
+
+      _scheduleNotify: function() {
+        this._debouncer = Polymer.Debounce(this._debouncer, 
+          this._notify);
+        this._debouncer.context = this;
+        Polymer.dom.addDebouncer(this._debouncer);
+      },
+
+      _notify: function(mxns) {
+        var info = mxns || {
+          target: this.node,
+          addedNodes: this._addedNodes,
+          removedNodes: this._removedNodes
+        }
+        this._updateContentElements(info);
+        this._callListeners(info);
+        this._addedNodes = [];
+        this._removedNodes = [];
+      },
+
+      _notifyInitial: function(listener) {
+        var info = {
+          target: this.node,
+          addedNodes: this.domApi.childNodes,
+          removedNodes: []
+        };
+        this._callListener(listener, info);
+      },
+
+      _updateContentElements: function(info) {
+        this._observeContentElements(info.addedNodes);
+        this._unobserveContentElements(info.removedNodes);
+      },
+
+      _observeContentElements: function(elements) {
+        for (var i=0, h, n; (i < elements.length) && (n=elements[i]); i++) {
+          if (this._isContent(n)) {
+            n.__observeNodesMap = n.__observeNodesMap || new WeakMap();
+            if (n.__observeNodesMap.get(this) == null) {
+              h = this._observeContent(n);
+              n.__observeNodesMap.set(this, h);
+            }
+          }
+        }
+      },
+
+      _observeContent: function(content) {
+        return Polymer.dom(content).observeNodes(this._notify.bind(this));
+      },
+
+      _unobserveContentElements: function(elements) {
+        for (var i=0, n, h; (i < elements.length) && (n=elements[i]); i++) {
+          if (this._isContent(n)) {
+            h = n.__observeNodesMap.get(this);
+            if (h) {
+              n.__observeNodesMap.set(this, null);
+              Polymer.dom(n).unobserveNodes(h);
+            }
+          }
+        }
+      },
+
+      _isContent: function(node) {
+        return (node.localName === 'content');
+      },
+
+      _callListeners: function(info, filter) {
+        var o$ = this._listeners;
+        for (var i=0, o; (i < o$.length) && (o=o$[i]); i++) {
+          if (!filter || filter(o.options)) {
+            this._callListener(o, info);
+          }
+        }
+      },
+
+      _callListener: function(listener, info) {
+        return listener.fn.call(this.node, info);
+      }
+
+    };
+
+    if (Settings.useShadow) {
+
+      var ensureSetup = DomApi.Mutation.prototype._ensureSetup;
+
+      Polymer.Base.extend(DomApi.Mutation.prototype, {
+        
+        _ensureSetup: function(options) {
+          this._trackAttributes = this._trackAttributes ||

```

Comment:
Impractical in shadow? Remove?  Otherwise, note that _trackAttributes tracking is coarse (ideally flag is cleared when last callback with attributes:true is removed.